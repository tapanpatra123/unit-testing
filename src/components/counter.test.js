import React from 'react';
import { shallow, } from 'enzyme';
import Counter from './counter'
import { shallowToJson} from 'enzyme-to-json';

describe("Counter page",()=>{

    let wrapper;
    const props ={
        name : "Title Goes Here"
    }
    
    beforeEach(()=>{
        wrapper = shallow(<Counter {...props}/>)
    })

    it("to snapshot",()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it("Receive props Name",()=>{
        const gotoText = wrapper.find("h1").text();
        expect(gotoText).toBe("Title Goes Here");
    })

    it("render the Default value of counter",()=>{
        const count = wrapper.find("h2");
        expect(count.text()).toBe('0');
    })
    
    it("render the click event of increment button and increment counter value",()=>{
        const buttoninCrement = wrapper.find('.increament');
        buttoninCrement.simulate('click');
        const count = wrapper.find("h2");
        expect(count.text()).toBe('1');
    })

    it("render the click event of decrrment button and decrement counter value",()=>{

        const buttoninCrement = wrapper.find('.increament');
        buttoninCrement.simulate('click');
        expect(wrapper.find("h2").text()).toBe('1');

        const buttonDecreament = wrapper.find('.decreament');
        buttonDecreament.simulate('click');
        expect(wrapper.find("h2").text()).toBe('0');
    })

    // it("")

})