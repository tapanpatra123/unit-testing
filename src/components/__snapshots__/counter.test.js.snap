// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter page to snapshot 1`] = `
<div>
  <h1>
    Title Goes Here
  </h1>
  <button
    className="increament"
    onClick={[Function]}
  >
     Click To Increase 
  </button>
  <button
    className="decreament"
    onClick={[Function]}
  >
     Click To Decrease 
  </button>
  <h2>
    0
  </h2>
</div>
`;
